Automattic/wp-calypso
# Subscriptions Management: Create Unfollow button

Closes https://github.com/Automattic/wp-calypso/issues/74987

## Proposed Changes

* Moves the `NotificationSettings` component to a more generic/reusable component called `SettingsPopover`
  * It can be used on the future Comments tab, to display comment settings on a popover
* Creates `SiteSettings` component and apply to the `SiteRow`
* Create `UnfollowSiteButton` to be used within the `SiteSettings`
  * For the hover/focus state we are using the design suggested here: p1680120736416249-slack-C02TCEHP3HA
* Create `UnfollowIcon`, which is a SVG, but as React component, to make it easier to apply CSS styles using `fill`

## Testing Instructions

1. Run this PR on your local env
2. Go to http://calypso.localhost:3000/subscriptions/sites
     Make sure you have a valid `subkey` cookie on calypso.localhost:3000 domain
3. Click on the ellipsis icon and verify if the popover is rendered as expected

<img width="508" alt="Screenshot 2023-03-30 at 17 49 53" src="https://user-images.githubusercontent.com/3113712/228970788-173ba304-b6c6-475b-85d3-2f4506f7b16d.png">

## Pre-merge Checklist
